internal/batches/service/remote.go

```diff
@@ -96,6 +102,121 @@ func (svc *Service) CreateBatchSpecFromRaw(
 	return resp.CreateBatchSpecFromRaw.ID, nil
 }
 
+// UploadBatchSpecWorkspaceFile uploads workspace files to the server.
+func (svc *Service) UploadBatchSpecWorkspaceFile(workingDir string, batchSpecID string, steps []batches.Step) error {
+	for _, step := range steps {
+		for _, mount := range step.Mount {
+			if err := svc.handleWorkspaceFile(workingDir, mount.Path, batchSpecID); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
```

Comment:
I think this has to deduplicate in the following case where the same thing is mounted to multiple steps:

```yaml
steps:
  - run: asdf
    mounts: [{tmp:/work/tmp}]
  - run: asdf 2
    mounts: [{tmp:/work/tmp}]
```